invenio_assets/__init__.py

```diff
@@ -30,20 +24,120 @@
     >>> assets = InvenioAssets(app)
 
 During initalization two Flask extensions :class:`flask_assets.Environment` and
-:class:`flask_collect.Collect` are instantiated and configured. Also bundles
-specified in the entry point group called ``invenio_assets.bundles`` are
-:ref:`automatically registered <entry-points-loading>`.
+:class:`flask_collect.Collect` are instantiated and configured.
+
+In order to build the assets, you can choose between using:
+
+* :ref:`Flask-WebpackExt <use-flask-webpackext>`;
+* :ref:`Flask-Assets <use-flask-assets>` (it will be deprecated soon.
+  See the
+  `blog post <https://inveniosoftware.org/blog/invenio-v300-released/>`_
+  for more information).
+
+Bundles specified in the entry point groups called
+``invenio_assets.webpack`` (if you are using Flask-WebpackExt) and
+``invenio_assets.bundles`` (if you are using Flask-Assets) are
+automatically registered by Invenio-Assets.
+
+.. _use-flask-webpackext:
+
+Using Flask-WebpackExt
+----------------------
 
 Bundles
--------
-The Flask-Assets package provides (via webassets) a class
+~~~~~~~
+The `Flask-WebpackExt <https://flask-webpackext.readthedocs.io/en/latest/>`_
+package provides a class
+:class:`flask_webpackext.project.WebpackBundle` for declaring the needed
+assets and NPM dependencies of each one of your modules.
+
+.. code-block:: python
+
+     from flask_webpackext import WebpackBundle
+     bundle1 = WebpackBundle(
+         __name__,
+         './modules/module1/static',
+         entry={
+             'module1-app': './js/module1-app.js',
+         },
+         dependencies={
+             'jquery': '^3.2.1'
+         }
+     )
+
+The NPM dependencies defined in the bundles will be used to generate the
+``package.json`` file.
+
+Entry points loading
+++++++++++++++++++++
+
+Invenio-Assets will automatically load bundles defined by the entry point
+group ``invenio_assets.webpack``.
+
+Example:
+
+.. code-block:: python
+
+   # setup.py
+   setup(
+       # ...
+       entry_points={
+           'invenio_assets.webpack': [
+               'mybundle = mypackage.bundles:mybundle',
+           ],
+           # ...
+        }
+        # ...
+    )
+
+Command-line interface
+~~~~~~~~~~~~~~~~~~~~~~
+
+We can now build the assets:
+
+.. code-block:: console
+
+    $ flask webpack buildall
+
+The command will copy all files from the src folder to the application
+instance folder designated for the Webpack project, download the npm packages
+and run Webpack to build our assets.
+
+Alternatively, we can execute each build step separately with the following
+flask webpack commands:
+
+    * ``create`` - Copy all sources to the working directory.
+    * ``build`` - Run npm install command and download all dependencies.
+    * ``install`` - Run npm run build and execute what you have defined in the
```

Comment:
Maybe we should stay high level here and just mention that this builds the assets since people won't have access to the `package.json`, what do you think?